pulp_rpm/app/depsolving.py

```diff
@@ -869,7 +869,10 @@ def run_solver_jobs(jobs):
                 # The solver is simply ignoring the problems encountered and proceeds associating
                 # any new solvables/units. This might be reported back to the user one day over
                 # the REST API.
-                logger.debug('Encountered problems solving: {}'.format(', '.join(problems)))
+                logger.warning(
```

Comment:
I agree that it should be higher than a debug level.
I don't know if it should be info or warning. The potential concern is if it happens often naturally (it's best effort and some dependencies might not be resolved) , then we'll create a lot of warnings for users, and they potentially can't do much about it? or can they?
Is there anything a user can do to eliminate the warning?
I guess I'm trying to understand what kind of warnings one can see here.

Reply:
Honestly, part of me wants to make all "best-effort" behavior go behind an opt-in flag.  It seems to me like it would be better if we defaulted to failing the task on encountering any problem.

Reply:
It would at least make issue discovery easier

Reply:
That sounds good, more explicit.
What do you mean by all best-effort behaviour? for RPM depsolving or for all-all? :) Advisories as well?
E.g. in EPEL repo there are always advisories which refer to a no-longer-existing packages in a repo.

Reply:
There's not really any way to make a distinction, the dependency solver doesn't work with errata it just takes the combined list of RPMs, modules and module-defaults that come out of "child detection".  I think in that case if you were copying one of those errata you would always need to have the "best_effort" flag enabled, which I suppose means it would need to be Katello's default.

Reply:
@goosemania To continue with answering your question:

When an RPM gets copied without all of the dependencies present they'd see warnings like 

```Encountered problems solving dependencies: Nothing provides pkgX, Nothing provides pkgY, ...etc
```

And there'd be one message per RPM had dependency solving problems.

Reply:
Thanks!

Reply:
Is it possible to tell which RPM has issues? It seems that the message shows only which deps are missing.
I wonder if it's useful for those who would like to skip those rpms/remove them from copy query and not have warnings (or task failures if we default to a strict mode and not a best-effort one)

Reply:
It's kind of a pain to do that unfortunately.  It's likely possible but kind of messy